cpp/src/cluster/detail/kmeans_balanced.cuh

```diff
@@ -1042,9 +1042,10 @@ void build_hierarchical(const raft::resources& handle,
   IdxT n_mesoclusters = std::min(n_clusters, static_cast<IdxT>(std::sqrt(n_clusters) + 0.5));
   RAFT_LOG_DEBUG("build_hierarchical: n_mesoclusters: %u", n_mesoclusters);
 
-  // TODO: Remove the explicit managed memory- we shouldn't be creating this on the user's behalf.
```

Comment:
Please don't remove this line. We do still need to fix this, but it's going to be a longer-term thing.

Reply:
Fix it by replacing the array allocations with `make_managed_vector` as mentioned [here](https://github.com/rapidsai/raft/pull/2570#discussion_r1943334811)? Or do you have something else in mind?

Reply:
I mean just simply to leave the TODO in place. THough your change improves things, this is still a problem, even with your change. We shouldn't be forcing managed memory on users EVER. They should be full control over it. This was one spot that wasn't trivial to fix, and I'd like to still make sure it's marked as future tech debt and not remove the TODO.

Reply:
Done in https://github.com/rapidsai/cuvs/pull/659/commits/0e881e48d48e2e2068ba005ba6d9f10980cdfb45 .